src/OrbitQt/AnnotatingSourceCodeHandler.cpp

```diff
@@ -0,0 +1,178 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "AnnotatingSourceCodeHandler.h"
+
+#include <chrono>
+#include <filesystem>
+
+#include "CodeReport/AnnotateDisassembly.h"
+#include "CodeReport/AnnotatingLine.h"
+#include "ObjectUtils/ElfFile.h"
+#include "OrbitBase/ImmediateExecutor.h"
+#include "OrbitBase/ReadFileToString.h"
+#include "OrbitBase/Result.h"
+#include "QtUtils/MainThreadExecutorImpl.h"
+#include "SourcePathsMapping/MappingManager.h"
+#include "SourcePathsMappingUI/AskUserForFile.h"
+
+namespace orbit_qt {
+AnnotatingSourceCodeHandler::AnnotatingSourceCodeHandler(
+    const QPointer<orbit_code_viewer::OwningDialog>& dialog,
```

Comment:
maybe CHECK(dialog != nullptr) ? Or would this class work with a nullptr?

Reply:
done.